in_toto/in_toto_record.py

```diff
@@ -14,166 +13,141 @@
-               "[--verbose]\n"
-      "Commands:\n{0}"
-               "start [--materials <filepath>[ <filepath> ...]]\n{0}"
-               "stop  [--products <filepath>[ <filepath> ...]]\n"
-               .format(lpad))
-
-  in_toto_args = parser.add_argument_group("in-toto options")
   # FIXME: Do we limit the allowed characters for the name?
-  in_toto_args.add_argument("-n", "--step-name", type=str, required=True,
-      help="Unique name for link metadata")
+  parent_parser.add_argument("-n", "--step-name", type=str, required=True,
+      help="Unique name for link metadata", metavar="<unique step name>")
+
+  # Either a key or a gpg key id have to be specified but not both
+  key_args_group = parent_parser.add_mutually_exclusive_group(required=True)
+  key_args_group.add_argument("-k", "--key", type=str,
+      help="Path to private key to sign link metadata (PEM)",
+      metavar="<signing key path>")
+  key_args_group.add_argument("-g", "--gpg", nargs="?", const=True,
+      metavar="<gpg keyid>", help=("GPG keyid to sign link metadata "
+      "(if set without argument, the default key is used)"))
+
+  parent_parser.add_argument("--gpg-home", dest="gpg_home", type=str,
+      help="Path to GPG keyring (if not set the default keyring is used)",
+      metavar="<gpg keyring path>")
 
-  in_toto_args.add_argument("-k", "--key", type=str, required=True,
-      help="Path to private key to sign link metadata (PEM)")
+  parent_parser.add_argument("-v", "--verbose", dest="verbose",
+      help="Verbose execution.", default=False, action="store_true")
 
-  in_toto_args.add_argument("-v", "--verbose", dest='verbose',
-      help="Verbose execution.", default=False, action='store_true')
+  subparser_start = subparsers.add_parser("start", parents=[parent_parser])
+  subparser_stop = subparsers.add_parser("stop", parents=[parent_parser])
 
   subparser_start.add_argument("-m", "--materials", type=str, required=False,
-      nargs='+', help="Files to record before link command execution")
+      nargs='+', help="Files to record before link command execution",
+      metavar="<material path>")
 
   subparser_stop.add_argument("-p", "--products", type=str, required=False,
-      nargs='+', help="Files to record after link command execution")
+      nargs='+', help="Files to record after link command execution",
+      metavar="<product path>")
 
   args = parser.parse_args()
 
   # Turn on all the `log.info()` in the library
   if args.verbose:
-    log.logging.getLogger().setLevel(log.logging.INFO)
+    in_toto.log.logging.getLogger().setLevel(in_toto.log.logging.INFO)
 
   # Override defaults in settings.py with environment variables and RCfiles
   in_toto.user_settings.set_settings()
 
+  # If `--gpg` was set without argument it has the value `True` and
+  # we will try to sign with the default key
+  gpg_use_default = (args.gpg == True)
+
+  # Otherwise we interpret it as actual keyid
+  gpg_keyid = None
+  if args.gpg != True:
```

Comment:
should this be
```python
if not gpg_use_default and args.gpg:
  ...
```
or so?

Reply:
Works either way, but yours is definitely more readable. I'll update and write a better comment.